File: python/consumer.py
from confluent_kafka import Consumer

consumer = Consumer({
    'bootstrap.servers': 'localhost:9092',
    'group.id': 'test-app'
})

consumer.subscribe(['test_topic'])

while True:
    msg = consumer.poll(1.0)

    if msg is None:
        continue
    if msg.error():
        print("Consumer error: {}".format(msg.error()))
        continue

    print('Received message: {}'.format(msg.value().decode('utf-8')))

consumer.close()

File: javascript/src/consumer.ts
import { Kafka, EachMessagePayload } from 'kafkajs';
import config from './config.json';

const client = new Kafka({
  clientId: 'test-app',
  brokers: [config.broker]
});

const processMessage = async ({ topic, partition, message}: EachMessagePayload) => {
  console.log(`Received a message from ${topic} [${partition}]: ${message.value.toString()}`)
}

async function consumeMessages(kafkaClient: Kafka) {
  // creste consumer
  const consumer = kafkaClient.consumer({ groupId: 'test-group' });
  await consumer.connect();

  // subscribe to a topic
  await consumer.subscribe({
    topic: 'test_topic',
    fromBeginning: true,
  });

  await consumer.run({
    eachMessage: processMessage,
  })

}

// RUN
(async() => {
  await consumeMessages(client);
})();